frame/nomination-pools/src/lib.rs

```diff
@@ -582,25 +583,237 @@ pub struct PoolRoles<AccountId> {
+					// the total durations passed since the last commission update.
+					let blocks_passed = <frame_system::Pallet<T>>::block_number().saturating_sub(p);
+
+					// calculate intervals passed since last `last_updated`
+					let intervals_passed = if blocks_passed == Zero::zero() {
+						Zero::zero()
+					} else {
+						blocks_passed.div(t.min_delay).saturated_into::<u32>()
+					};
+
+					// the maximum allowed increase, where the current `max_increase` Perbill is
+					// converted into a u32 by multiplying itself with 100_u32 and multiplied by
+					// durations passed, before being converted back into a Perbill.
+					let max_allowed_increase =
+						Perbill::from_percent((t.max_increase * 100_u32) * intervals_passed);
+
+					// throttled (true) if attempted increase is greater than the maximum allowed
+					// increase, as a percentage.
+					attempted_increase > max_allowed_increase
+				},
+			) {
+				return true
+			}
+		}
+		false
+	}
+
+	/// Set the pool's commission.
+	///
+	/// Update commission based on `current`. If a `None` current value is supplied, allow the
+	/// commission to be removed in all cases, without any throttling restrictions. If `change_rate`
+	/// is present, record the current block  as the previously updated commission. If the supplied
+	/// commission is zero, `None` will be  inserted and `payee` will be ignored.
+	fn maybe_update_current(
+		&mut self,
+		current: &Option<(Perbill, T::AccountId)>,
+	) -> DispatchResult {
+		if current.is_none() {
+			self.current = None;
+		} else {
+			let (commission, payee) = current.as_ref().ok_or(Error::<T>::NoCommissionSet)?;
+			ensure!(!self.throttling(&commission), Error::<T>::CommissionChangeThrottled);
+			ensure!(
+				self.max.map_or(true, |m| commission <= &m),
+				Error::<T>::CommissionExceedsMaximum
+			);
+			self.current = Some((*commission, payee.clone()));
+		}
+		let _ = self.register_update(<frame_system::Pallet<T>>::block_number());
+		Ok(())
+	}
+
+	/// Set the pool's maximum commission.
+	///
+	/// The pool's maximum commission can initially be set to any value, below global maximum if
+	/// set, and only smaller values thereafter. If larger values are attempted, this function will
+	/// return a dispatch error.
+	///
+	/// If `current.0` is larger than an updated max commission value, then `current.0` will also be
+	/// updated to the new maximum. This will also register a `last_updated` update to the
+	/// commission.
+	fn maybe_update_max(&mut self, new_max: Perbill) -> DispatchResult {
+		if let Some(old) = self.max.as_mut() {
+			if new_max > *old {
+				return Err(Error::<T>::MaxCommissionRestricted.into())
+			}
+			*old = new_max;
+		} else {
+			self.max = Some(new_max)
+		};
+		let updated_current = self
+			.current
+			.as_mut()
+			.map(|(x, _)| {
+				let u = *x > new_max;
+				*x = (*x).min(new_max);
+				u
+			})
+			.unwrap_or(false);
+
+		if updated_current {
+			self.register_update(<frame_system::Pallet<T>>::block_number());
```

Comment:
@Ank4n  when a maximum commission change pushes a current commission down, `last_updated` is also updated.